presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -195,4 +245,41 @@ private void initializeColumnReaders()
             columnReadersMap.put(column, ParquetColumnReader.createReader(column));
         }
     }
+
+    private Block readBlock(String name, Type type, List<String> path, IntList offsets)
+            throws IOException
+    {
+        path.add(name);
```

Comment:
@nezihyigitbasi I am using path variable to recursively process nested path names. Need to add new path components when getting into a deeper level, and remove it when deeper level is done, comes back to the current level. Yes, it is a single thread. Let me see whether I could make it more clear